kong/runloop/handler.lua

```diff
@@ -1206,37 +1207,38 @@ return {
         or redirect_status_code == 307
         or redirect_status_code == 308
         then
-          header["Location"] = "https://" .. forwarded_host .. var.request_uri
+          header["Location"] = "https://" .. forwarded_host .. ctx.request_uri
           return kong.response.exit(redirect_status_code)
         end
       end
 
-      -- mismatch: non-http/2 request matched grpc route
-      if (protocols and (protocols.grpc or protocols.grpcs) and http_version ~= 2 and
-        (content_type and sub(content_type, 1, #"application/grpc") == "application/grpc"))
-      then
-        return kong.response.exit(426, { message = "Please use HTTP2 protocol" }, {
-          ["connection"] = "Upgrade",
-          ["upgrade"]    = "HTTP/2",
-        })
-      end
+      if protocols.grpc or protocols.grpcs then
+        -- perf: branch usually not taken, don't cache var outside
+        local content_type = var.content_type
+
+        if content_type and find(content_type, "application/grpc", 1, true) == 1 then
```

Comment:
I think the original version of using `sub(content_type, 1, #"application/grpc") == "application/grpc"` would be faster than `string.find`.

Reply:
I did a quick test, sub is indeed (slightly) faster than find but are in the noise margin. Consider this branch
is only taken in grpc, i will revert it back to sub to reduce diff from original.

Reply:
perhaps "slower" but does the sub generate more garbage?

Reply:
no, there's some escape analysis and operation folding.  if JITted, the substring is never created, the comparison is made in-place.